src/coreclr/jit/async.cpp

```diff
@@ -1100,6 +1098,50 @@ void Async2Transformation::Transform(
     m_resumptionBBs.push_back(resumeBB);
 }
 
+GenTreeCall* Async2Transformation::CreateAllocContinuationCall(AsyncLiveness& life, GenTree* prevContinuation, unsigned gcRefsCount, unsigned dataSize)
+{
+    GenTree* gcRefsCountNode = m_comp->gtNewIconNode((ssize_t)gcRefsCount, TYP_I_IMPL);
+    GenTree* dataSizeNode = m_comp->gtNewIconNode((ssize_t)dataSize, TYP_I_IMPL);
+    // If VM requests that we report the method handle, or if we have a shared generic context method handle
+    // that is live here, then we need to call a different helper.
+    GenTree* methodHandleArg = nullptr;
```

Comment:
If I read this correctly we will use an alternative continuation allocators unconditionally, if we have generic contexts and allocator will sort it out if there is an object to hold on to.  

I guess we cannot know statically if we will have loader allocator or not. Not a big deal, I think.

Reply:
Yes, I think all shared generics need to go through these new  helpers. We cannot know if the generic context was instantiated with an unloadable type during JIT.

For the `continuationsNeedMethodHandle` case I assume the loader allocator will always be non-null (since it's only set when the ALC is collectible).